transactionparser/src/channelchanges.js

```diff
@@ -0,0 +1,86 @@
+const normalizeNodes = require('./utils').normalizeNodes;
+const dropsToXRP = require('./utils').dropsToXRP;
+const BigNumber = require('bignumber.js');
+
+function parsePaymentChannelStatus(node) {
+  if (node.diffType === 'CreatedNode') {
+    return 'created'
+  }
+
+  if (node.diffType === 'ModifiedNode') {
+    return 'modified'
+  }
+
+  if (node.diffType === 'DeletedNode') {
+    return 'deleted'
+  }
+  return undefined
+}
+
+function summarizePaymentChannel(node) {
+
+      const final = (node.diffType === 'CreatedNode') ? node.newFields : node.finalFields;
+      const prev = node.previousFields || {};
+
+      return {
+        // Status may be 'created', 'modified', or 'deleted'.
+        status: parsePaymentChannelStatus(node),
+
+        // The LedgerIndex indicates the Channel ID, which is necessary to sign claims.
+        channelId: node.ledgerIndex,
+
+        // The source address that owns this payment channel.
+        // This comes from the sending address of the transaction that created the channel.
+        source: final.Account,
+
+        // The destination address for this payment channel.
+        // While the payment channel is open, this address is the only one that can receive
+        // XRP from the channel. This comes from the Destination field of the transaction
+        // that created the channel.
+        destination: final.Destination,
+
+        // The change in the number of XRP drops allocated to this channel.
+        // This is positive if this is a PaymentChannelFund transaction.
+        channelAmountChangeDrops: prev.Amount ? 
+          new BigNumber(final.Amount)
+          .minus(new BigNumber(prev.Amount || 0))
+          .toString(10) :
+          undefined,
+
+        // The change in the number of XRP drops already paid out by the channel.
+        channelBalanceChangeDrops: final.Balance ? 
```

Comment:
@intelliot 
`channelBalanceChangeDrops: final.Balance ?`

I meant this line, with the conditional

Generally (some exceptions), with canonical metadata, final fields will have ALL the ledger entries' fields. Previous fields includes only those that have changed. 

So the question is when will `final.Balance` there actually be falsey ?

Reply:
@intelliot 

Ah, right of course, that makes sense. I got confused :)

Because they will be in the new fields right?

Reply:
@sublimator Ah, I see what you're saying - this should actually be:
```
channelBalanceChangeDrops: prev.Balance ?
```

So that the field is undefined if this transaction did not modify the channel's balance. I think you're right.

Reply:
Fixed in https://github.com/ripple/ripple-lib-extensions/commit/c92dd6ad6649cfe714bee6b3f7e1ad4bbfa56910